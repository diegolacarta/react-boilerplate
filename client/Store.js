import {observable} from 'mobx'

export default class Store {
  constructor(state = {todos: []}) {
    this.todos = observable(state.todos)
  }

  addTodo(todo) {
    this.todos.push(todo)
  }

  fetch() {
    return new Promise((resolve) => {
      setTimeout(() => {
        this.todos.push('my async todo')
        resolve()
      }, 1000)
    })
  }
}