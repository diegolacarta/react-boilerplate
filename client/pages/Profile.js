import React from 'react'
import {inject, observer} from 'mobx-react'
import {Link} from 'react-router'

@inject('store')
@observer
export default class Profile extends React.Component {
  static propTypes = {
    store: React.PropTypes.object
  }

  static fetchData({store}) {
    return store.fetch()
  }

  onKeyDown = (event) => {
    if (event.keyCode === 13) {
      this.props.store.addTodo(event.target.value)
    }
  }

  render() {
    return (
      <div>
        <h1>This is profile asdf</h1>
        <input type="text" onKeyDown={this.onKeyDown}/>
        <ul>
          {this.props.store.todos.map((todo, index) => (
            <li key={index}>{todo}</li>
          ))}
        </ul>
        <Link to="/about">Go to about</Link>
      </div>
    )
  }
}