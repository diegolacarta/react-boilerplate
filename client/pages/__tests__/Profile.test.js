import React from 'react'
import renderer from 'react-test-renderer'
import Profile from '../Profile'
import Store from '../../Store'

test('Profile', () => {
  const tree = renderer.create(
    <Profile store={new Store()}/>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})