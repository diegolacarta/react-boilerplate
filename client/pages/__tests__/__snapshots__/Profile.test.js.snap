// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile 1`] = `
<div>
  <h1>
    This is profile
  </h1>
  <a
    onClick={[Function]}
    style={Object {}}
  >
    Go to about
  </a>
  <div>
    <input
      onKeyDown={[Function]}
      type="text"
    />
    <ul>
      <li>
        <span>
          fetched todo
        </span>
        <button
          onClick={[Function]}
        >
          remove
        </button>
        <button
          onClick={[Function]}
        >
          up
        </button>
        <button
          onClick={[Function]}
        >
          down
        </button>
      </li>
    </ul>
  </div>
</div>
`;
